client/utils/assignments.js

```diff
@@ -1,26 +1,30 @@
 import {get, includes, isNil} from 'lodash';
-import {ASSIGNMENTS, PRIVILEGES, PLANNING} from '../constants';
+import {ASSIGNMENTS, PRIVILEGES} from '../constants';
 import {lockUtils, getCreator, getItemInArrayById, isExistingItem} from './index';
 
 const isNotLockRestricted = (assignment, session) => (
     !get(assignment, 'lock_user') ||
         lockUtils.isItemLockedInThisSession(assignment, session)
 );
 
-const isTextAssignment = (assignment) => (
-    get(assignment, 'planning.g2_content_type') === PLANNING.G2_CONTENT_TYPE.TEXT
-);
+const isTextAssignment = (assignment, contentTypes = []) => {
+    const textContentTypes = contentTypes.filter((c) => get(c, 'content item type') === 'text');
+
+    return textContentTypes.length === 0 ? get(assignment, 'planning.g2_content_type') === 'text' :
```

Comment:
This fails if:
`contentTypes = [{qcode: 'text'}, {qcode: 'new_text', 'content item type': 'text'}`
`assignment.planning.g2_content_type = 'text'`

Because `textContentTypes.length === 1`, but the entry for `text` does not have a `'content item type'` value assigned.
It might be better to get the entry in the CV by qcode, then use `lodash.get` to get the `'content item type'` and default to `qcode` instead.